### ISSUE TITLE
Coder plugin: clean up error-handling logic

### ISSUE THREAD

**Issue by Parkreiner** [coder/backstage-plugins#134]
The Coder plugin has some vestigial code from before we started using Axios. Now that we have Axios's `AxiosError`, there's far less of a need for the custom `BackstageHttpError`

In addition, the logic for spying on potentially-expired Coder session tokens can be cleaned up. The current code works, but when a Coder query fails, the plugin will make an entirely brand new request to revalidate the auth state. But if the error has a 401 status, we don't need to do that – we know that the session token is likely invalid, so we can mutate the query cache directly.

Tried to get these fixed really quickly in #132, but it started making the tests fail, so I decided to split this off into a separate issue.
